torch/distributed/distributed_c10d.py

```diff
@@ -2732,7 +2732,7 @@ def all_gather(tensor_list, tensor, group=None, async_op=False):
 
 
 @_exception_logger
-def all_gather_into_tensor(output_tensor, input_tensor, group=None, async_op=False):
+def all_gather_into_tensor(output_tensor, input_tensor, group: ProcessGroup = None, async_op=False):
```

Comment:
I matched the type in the doc below but lmk if this is wrong

Reply:
I don't think we should add a single ProcessGroup type to this API, the other args like `output_tensor` and `input_tensor` does not have type too

Reply:
The heredoc below specify this

```
Args:
        output_tensor (Tensor): Output tensor to accommodate tensor elements
            from all ranks. It must be correctly sized to have one of the
            following forms:
            (i) a concatenation of all the input tensors along the primary
            dimension; for definition of "concatenation", see ``torch.cat()``;
            (ii) a stack of all the input tensors along the primary dimension;
            for definition of "stack", see ``torch.stack()``.
            Examples below may better explain the supported output forms.
        input_tensor (Tensor): Tensor to be gathered from current rank.
            Different from the ``all_gather`` API, the input tensors in this
            API must have the same size across all ranks.
        group (ProcessGroup, optional): The process group to work on. If None,
the default process group will be used.
```

I'm cool with taking it out but wondering if there's a reason why this method shouldnt have type annotations